lib/db.js

```diff
@@ -692,7 +692,7 @@ module.exports = (
               // Ensure that we don't return lastAccessTime if redis is down
               isRedisOk = false
             }
-            return unpackTokensFromRedis(result)
+            return this.safeUnpackTokensFromRedis(uid, result)
```

Comment:
There are also `unpackTokensFromRedis` on [lines 596](https://github.com/mozilla/fxa-auth-server/blob/855d6814ec1df26832afda58dc4e242cc68f9e4a/lib/db.js#L596), [668](https://github.com/mozilla/fxa-auth-server/blob/855d6814ec1df26832afda58dc4e242cc68f9e4a/lib/db.js#L596), and [1350](https://github.com/mozilla/fxa-auth-server/blob/855d6814ec1df26832afda58dc4e242cc68f9e4a/lib/db.js#L1350) do those calls need to be converted too?

Reply:
I thought about this pretty carefully.

Those calls all occur during updates and I think we still want the endpoints that call them to fail if they encounter this situation, so there's less benefit to returning an empty value there. And while you could argue that we should be more proactive about clobbering the data in those cases, if we add this handling to them then they wouldn't show up in Sentry any more so it might be easier for us to miss them if they do occur.

Given that the one case of this behaviour we've seen so far occurred during a get, and that gets are more frequent than updates, it felt like a sufficiently good fix to only apply it to those operations. If it ever becomes an issue on the updating endpoints we can make a conscious decision to add this logic to them then, although given the frequency of this error up to now I'm hopeful that we'll just never see it again.

Reply:
Solid reasoning.